### PR TITLE
[14.0][ADD] stock_valuation_layer_total_value: add to 14.0

### PR DESCRIPTION
**Pull request by JuanyDForgeflow**
This module show the value of the original valuation layer after doing
a landed cost.

### CODE REVIEW

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        string="New value",
+        compute="_compute_original_layer_values",
+    )
+
+    def _compute_original_layer_values(self):
+        original_value = ad_unit_price = 0
+        for rec in self:
+            if len(rec.stock_valuation_layer_ids):
+                children_value = sum(rec.stock_valuation_layer_ids.mapped("value"))
+                original_value = rec.value / children_value
```
Comment (AaronHForgeFlow) [inline stock_valuation_layer_total_value/models/stock_valuation_layer.py]: A ZeroDivisionError seems possible here

File: stock_valuation_layer_total_value/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+To use this module you need to:
```
Comment (AaronHForgeFlow) [inline stock_valuation_layer_total_value/readme/USAGE.rst:1]: Can you complete this or delete it?

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class StockValuationLayer(models.Model):
+    _inherit = "stock.valuation.layer"
+
+    original_value = fields.Float(
+        string="Original value",
+        compute="_compute_original_layer_values",
+    )
+    adjusted_unit_price = fields.Float(
```
Comment (AaronHForgeFlow) [inline stock_valuation_layer_total_value/models/stock_valuation_layer.py]: I don't see how a field called "adjusted_unit_price" shows as string "New Value", should it be "new unit price"?

**Review by AaronHForgeFlow — COMMENTED** [review]
As this is done for landed costs, shouldn't the module name contain landed cost on it?

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply stock_valuation_layer_total_value/models/stock_valuation_layer.py]: @AaronHForgeFlow there can be other reasons like inventory revaluations.

Comment (AaronHForgeFlow) [reply stock_valuation_layer_total_value/models/stock_valuation_layer.py]: Ok, but in that case we should we remove the dependency to stock_landed_cost. If seems to me it is only used for tests.

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py (hunk 3 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/models/stock_valuation_layer.py]: @AaronHForgeFlow   I changed the name

File: stock_valuation_layer_total_value/readme/USAGE.rst (hunk 2 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/readme/USAGE.rst:1]: @AaronHForgeFlow it's complete now

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py (hunk 1 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/models/stock_valuation_layer.py]: @AaronHForgeFlow I removed the dependency to stock_landed_cost

**Comment by JordiBForgeFlow** [comment]
@JuanyDForgeflow I just created a layer, and the fields 'original value' and 'new value' appear as 0. Does this make sense?

image

**Comment by JordiBForgeFlow** [comment]
@JuanyDForgeflow I used the runboat to test. I created a product with FIFO costing and cost 10, and then updated the quantity on hand to 1.

File: stock_valuation_layer_total_value/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+This module show the value of the original valuation layer after doing
```
Comment (rousseldenis) [inline stock_valuation_layer_total_value/readme/DESCRIPTION.rst]:
```suggestion
This module shows the value of the original valuation layer after doing
```

File: stock_valuation_layer_total_value/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
```
Comment (rousseldenis) [inline stock_valuation_layer_total_value/__manifest__.py]: Coppy paste ? :smiley:

File: stock_valuation_layer_total_value/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock Valuation Layer Total Value",
+    "summary": "Show total value on tree and form view",
+    "version": "14.0.1.0.0",
+    "development_status": "Production/Stable",
```
Comment (rousseldenis) [inline stock_valuation_layer_total_value/__manifest__.py:7]: As first addition, maybe too soon?

File: stock_valuation_layer_total_value/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    "category": "stock",
+    "website": "https://github.com/OCA/stock-logistics-warehouse",
+    "author": "Forgeflow (OCA), Odoo Community Association (OCA)",
```
Comment (rousseldenis) [inline stock_valuation_layer_total_value/__manifest__.py]:
```suggestion
    "author": "Forgeflow, Odoo Community Association (OCA)",
```

No ?

File: stock_valuation_layer_total_value/test/test_stock_valuation_layer_total_value.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+        return in_move.with_context(svl=True)
+
+    def test_valuation_layer_values(self):
+        move1 = self._make_in_move(self.product1, 10, unit_cost=10, create_picking=True)
+        move2 = self._make_in_move(self.product1, 5, unit_cost=15, create_picking=True)
+        original_svl = move1.stock_valuation_layer_ids
+        new_svl = move2.stock_valuation_layer_ids
+        original_svl.write({"stock_valuation_layer_ids": [(4, new_svl.id)]})
+        original_svl._compute_original_layer_values()
```
Comment (rousseldenis) [inline stock_valuation_layer_total_value/test/test_stock_valuation_layer_total_value.py]: @JuanyDForgeflow Have you tested that original_svl is not void as coverage says function is not covered.

Moreover, calling it manually would say you should add a correct depends() decorator

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    )
+
+    @api.depends('stock_valuation_layer_ids')
+    def _compute_original_layer_values(self):
+        original_value = new_unit_price = 0
+        for rec in self:
+            if len(rec.stock_valuation_layer_ids):
+                children_value = sum(rec.stock_valuation_layer_ids.mapped("value"))
+                original_value = rec.value + children_value
+                new_unit_price = original_value / rec.quantity
```
Comment (JordiBForgeFlow) [inline stock_valuation_layer_total_value/models/stock_valuation_layer.py]: @JuanyDForgeflow you should control if rec.quantity = 0

File: stock_valuation_layer_total_value/readme/DESCRIPTION.rst (hunk 4 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/readme/DESCRIPTION.rst]: Done

File: stock_valuation_layer_total_value/__manifest__.py (hunk 5 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/__manifest__.py]: Done

File: stock_valuation_layer_total_value/__manifest__.py (hunk 6 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/__manifest__.py:7]: Done

File: stock_valuation_layer_total_value/__manifest__.py (hunk 7 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/__manifest__.py]: Done

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py (hunk 9 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/models/stock_valuation_layer.py]: Done

File: stock_valuation_layer_total_value/test/test_stock_valuation_layer_total_value.py (hunk 8 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/test/test_stock_valuation_layer_total_value.py]: @rousseldenis Thanks the dependency was missing, now it's fine

File: stock_valuation_layer_total_value/models/stock_valuation_layer.py (hunk 9 above)
Comment (JuanyDForgeflow) [reply stock_valuation_layer_total_value/models/stock_valuation_layer.py]: @JordiBForgeFlow please test again

**Pull request closed without merging**